nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -1,22 +1,64 @@
 package com.uber.nullaway;
 
+import static com.uber.nullaway.NullabilityUtil.castToNonNull;
+
+import com.google.common.base.Preconditions;
 import com.google.errorprone.VisitorState;
+import com.google.errorprone.suppliers.Supplier;
+import com.google.errorprone.suppliers.Suppliers;
 import com.google.errorprone.util.ASTHelpers;
 import com.sun.source.tree.AnnotatedTypeTree;
 import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.AssignmentTree;
+import com.sun.source.tree.NewClassTree;
 import com.sun.source.tree.ParameterizedTypeTree;
 import com.sun.source.tree.Tree;
+import com.sun.source.tree.VariableTree;
 import com.sun.tools.javac.code.Attribute;
 import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.code.TypeMetadata;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 /** Methods for performing checks related to generic types and nullability. */
 public final class GenericsChecks {
 
-  private GenericsChecks() {
-    // just utility methods
+  private static final String NULLABLE_NAME = "org.jspecify.annotations.Nullable";
```

Comment:
Eventually we may need to handle `@Nullable` type-use annotations other than jspecify.  but we can leave this for now